src/CarbonAware.DataSources/CarbonAware.DataSources.WattTime/src/WattTimeDataSource.cs

```diff
@@ -59,36 +59,57 @@ public async Task<IEnumerable<EmissionsData>> GetCarbonIntensityAsync(IEnumerabl
         }
         return result;
     }
-    
+
     /// <inheritdoc />
     public async Task<EmissionsForecast> GetCurrentCarbonIntensityForecastAsync(Location location)
     {
         this.Logger.LogInformation($"Getting carbon intensity forecast for location {location}");
+        using (var activity = Activity.StartActivity())
+        {
+            var balancingAuthority = await this.GetBalancingAuthority(location, activity);
+                var forecast = await this.WattTimeClient.GetCurrentForecastAsync(balancingAuthority); 
+                return ForecastToEmissionsForecast(forecast, location);
+        } 
+    }
 
+    /// <inheritdoc />
+    public async Task<EmissionsForecast> GetCarbonIntensityForecastAsync(Location location, DateTimeOffset generatedAt)
```

Comment:
```suggestion
    public async Task<EmissionsForecast> GetCarbonIntensityForecastAsync(Location location, DateTimeOffset requestedAt)
```

Given the log messages and implementation, would `requestedAt` be a more accurate variable name?  If yes, update here and throughout the method/interface.